src/applications/TrainingModule.cpp

```diff
@@ -2001,44 +2042,70 @@ std::vector<int> TrainingModule::SVMFFSBasedFeatureSelection(const VariableSizeM
         reducedFeatureSet(j, reducedFeatureSet.Cols() - 1) = (double)value / 10000;
       }
 
-
       //check crossvalidated performance after adding the current feature
       double bestCV = 0;
       double bestC = 1;
       double bestG = 1 / reducedFeatureSet.Cols();
-      if (classifiertype == 2)
+      if (optimizationtype == 1)
       {
-        for (double cValue = -5; cValue <= 5; cValue = cValue + 1)
-          for (double gValue = -5; gValue <= 5; gValue = gValue + 1)
+        if (classifiertype == 2)
+        {
+          for (double cValue = -5; cValue <= 5; cValue = cValue + 1)
+            for (double gValue = -5; gValue <= 5; gValue = gValue + 1)
+            {
+              VectorDouble result;
+              if (cvtype == 1)
+                result = InternalCrossValidationResubstitution(reducedFeatureSet, labels, pow(2, cValue), pow(2, gValue), classifiertype);
+              else
+                result = InternalCrossValidation(reducedFeatureSet, labels, pow(2, cValue), pow(2, gValue), classifiertype);
+
+              if (result[3] > bestCV)
+                bestCV = result[3];
+            }
+        }
+        else
+        {
+          for (double cValue = -5; cValue <=5; cValue = cValue + 1)
           {
-            VectorDouble result = InternalCrossValidationResubstitution(reducedFeatureSet, labels, pow(2, cValue), pow(2, gValue), classifiertype);
+            VectorDouble result;
+            if (cvtype == 1)
+              result = InternalCrossValidationResubstitution(reducedFeatureSet, labels, pow(2, cValue), 0.01, classifiertype);
+            else
+              result = InternalCrossValidation(reducedFeatureSet, labels, pow(2, cValue), 0.01, classifiertype);
+
```

Comment:
Please add comments. What are the hard coded values?

Reply:
for now, we have defined a fixed range to find optimal parameter values of the classifier. in future, we can ask the user to specify these ranges.

Reply:
added comments in the code